### DISCUSSION TITLE
High Sample Rate Issue Started With macOS Sonoma

### DISCUSSION THREAD

**Discussion by chris027** [ExistentialAudio/BlackHole#742] [General]
I use Blackhole for routing audio into a DSP app, which then routes audio out via the Merging Technologies AES67 driver. When I play audio at 24 bit / 48 kHz it works perfectly. Before Sonoma, audio at 24 bit / 352.8 kHz also worked perfectly, but now it doesn't. The audio is received by the application in the middle, but no audio is routed out. This happens with both applications I've tried. 

JRiver plays the 24 / 352.8 kHz music > Blackhole v5.0 is the output device > Audacity v3.3.3 is the app in the middle with Blackhole as its input device and outputting to > Merging AES67 driver as the output device. 

You can see in the screenshot below there is input, but no output on the meters in the upper right of Audacity. This only happens with high sample rate audio when Blackhole is used. Sending straight to the Merging driver works perfect. 

Not sure if Apple has done something that would set a sample rate limit on this stuff due to the new Mic Modes (even when not used at all) etc...

<img width="1315" alt="Screenshot 2023-10-07 at 6 42 29 PM" src="https://github.com/ExistentialAudio/BlackHole/assets/40768499/6513d6b4-ace9-4594-a7ba-e449e4d2e91a">

**Comment by chris027** [comment]
Just an FYI, the Merging AES67 driver is also a virtual interface. Not sure if passing audio from one virtual interface to another is now not allowed for high sample rates or something strange.

**Comment by chris027** [comment]
Ahhh. I got Audacity to work at 24/352.8 with Blackhole. Not sure what's going on now.

**Comment by bramses** [comment]
Hi @chris027, How did you get Blackhole working on Sonoma? I can't get it to work at all

**Reply by chris027** [reply]
It just works for me. Installed 14.1 update today and it still works great.

**Comment by eklyst** [comment]
ok so i dont understand half of the terms here but i think this is the same problem im having. if you could let me know if/when a solution is found that would be awesome

**Comment by devinroth** [comment]
Try uninstalling BlackHole and reinstalling it.

Devin

**Comment by trancephorm** [comment]
I have this same problem since I updated to Sonoma. Reinstalling Blackhole is not helping. More, I'm using some visual analyzer which is not working anymore, and it should receive input from Blackhole.

**Comment by devinroth** [comment]
Did you check volume and mute on BlackHole? Most of the time that’s the issue.

Devin

**Comment by trancephorm** [comment]
Yes, I checked everything multiple times... Sound that is going through Blackhole gets a bit garbled, sounds like sample rate is halved. Not saying it's because of Blackhole, I have several audio drivers here, but it used to work in Ventura. Also, I do have a problem with Vision 4X analyzer which just can't pick output from Blackhole (as its input), but system says there is a signal coming to Blackhole...

**Comment by devinroth** [comment]
Are you using a multi output? macOS messed that up in certain situations.

Devin

**Comment by trancephorm** [comment]
Yes, I am using Multi Output because I need to hear sound and to route it to Blackhole, so Vision 4X analyzer can listen to it. And then I use Screen Recorder to record graphic analysis together with audio, and it all used to work in Ventura.

**Comment by devinroth** [comment]
I guess the question is whether there is an issue with BlackHole or just with multi-outputs.

Try just using BlackHole to test if it works properly without a multi-output.

Devin

**Comment by trancephorm** [comment]
Just to let you know that this still doesn't work with Blackhole 0.6.0. The only way VISION 4X can receive audio for analysis is when it's run as VST, at least here it's such case.
